docs/iris/src/whitepapers/change_management.rst

```diff
@@ -0,0 +1,393 @@
+
+* when a new major version is **released**
+    check for new deprecation warnings, as for a minor release
+
+
+Details
+=======
+
+The Iris change process aims to minimise the negative effects of change, by
+providing :
+
+    * defined processes for release and change management
+    * release versioning
+    * backwards code compatibility through minor version releases
+    * a way to ensure compatibility with a new major version release
+    * deprecation notices and warnings to highlight all impending changes
+
+Our practices are intended be compatible with the principles defined in the
+`SemVer project <http://semver.org/>`_ .
+
+Key concepts covered here:
+    * :ref:`Release versions <iris_change_releases>`
+    * :ref:`Backwards compatibility <iris_backward_compatibility>`
+    * :ref:`Deprecation <iris_deprecations>`
+
+
+.. _iris_backward_compatibility:
+
+Backwards compatibility
+-----------------------
+
+"Backwards-compatible" changes are those that leave all possible existing API
+usages unchanged (see :ref:`terminology <iris_api>` below).
+Only such changes may be included in minor releases.
+
+the following are examples of backward-compatible changes :
+
+    * changes to documentation
+    * adding to a module : new submodules, functions, classes or properties
+    * adding to a class : new methods or properties
+    * adding to a function or method : new optional arguments or keywords
+
+The following are examples of **non-** backward-compatible changes :
+
+    * removing a call
+    * removing a module
+    * removing an object
+    * removing an object property
+    * removing a call argument or keyword
+    * renaming a module, object, property, call, argument or keyword
+    * adding a required argument
+    * removing a keyword (even one that has no effect)
+    * changing the effect of *any* particular combination of arguments and/or
+      keywords
+
+Note that it is also possible to modify the behaviour of an existing usage by
+making it depend on a newly-defined external control variable.  This is
+effectively a change to the 'default behaviour' of a specific usage.  Although
+this seems similar to adding a keyword, the cases where the new behaviour
+operates and where it does not are not distinguishable by a different code
+syntax, which makes this somewhat dangerous.  We do use this type of change,
+but any behaviour 'mode' controls of this sort are usually added as part of the
+:class:`iris.Future` definition.
+See :ref:`Usage of iris.FUTURE <iris_future_usage>`, below.
+
+
+.. _iris_api:
+
+Terminology : API, features, usages and behaviours
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The API is the components of the iris module and submodules which are
+"public" :  In Python, by convention, this means everything whose name does not
+have a leading underscore "_".
+This means all public modules and submodules, and their contained classes,
+public data and properties, functions and methods.
```

Comment:
I think this is an issue illustrated by https://github.com/SciTools/iris/pull/1962

That removed various public-named items from the iris.fileformats.grid module.
These, by my reckoning :
```CENTRE_TITLES, TIME_RANGE_INDICATORS, PROCESSING_TYPES, TIME_CODES_EDITION1, TIME_CODES_EDITION2, unknown_string, GribDataProxy```

Maybe that is ok, if they were excluded from the "__all__" in the original ?
It means their docstrings aren't published, and they are omitted from a "from iris.fileformats.grib import *"

Is that good enough to consider them ***private*** ??